### PR TITLE
use userland which; closes #3639

### LINKED ISSUE #3639: growl should use userland "which" package
We should consume userland modules for nontrivial operations to avoid maintenance burden.  continuation of my comments from #3542, which introduced a hand-rolled `which` implementation.

### PR DESCRIPTION
**Pull request by boneskull**
also refactor to use modern JS in `lib/growl.js`; update ESLint rules accordingly

### CODE REVIEW

**Comment by boneskull** [comment]
I'm going to ignore the coverage drop here.  not my fault!!!

File: lib/growl.js [hunk 1]
```diff
@@ -88,15 +95,11 @@ function display(runner) {
  * @private
  * @callback Growl~growlCB
  * @param {*} err - Error object, or <code>null</code> if successful.
- * @param {string} stdout - <code>stdout</code> from notification delivery
- *     process.
- * @param {string} stderr - <code>stderr</code> from notification delivery
- *     process. It will include timestamp and executed command with arguments.
  */
-function onCompletion(err, stdout, stderr) {
```
Comment (plroebuck) [inline lib/growl.js]: For what it's worth, I disagree with undocumenting the callback.

File: lib/growl.js [hunk 2]
```diff
@@ -23,8 +23,16 @@ var path = require('path');
...
  * @return {boolean} whether Growl notification support can be expected
  */
-exports.isCapable = function() {
-  return !process.browser && which(getSupportBinaries()) !== '';
+exports.isCapable = () => {
+  if (!process.browser) {
+    const binaries = getSupportBinaries();
+    if (binaries.length) {
+      try {
+        return Boolean(which(binaries, {all: true}));
```
Comment (plroebuck) [inline lib/growl.js]: This is broken now... `binaries` is an array, but your `which` takes a string necessitating a `for` loop.
You also only want the first match, so `all` is not as helpful as `nothrow` would have been.

**Comment by boneskull** [comment]
assuming this passes muster, I'll release v6.0.0-0 thereafter (maybe tomorrow?)

File: lib/growl.js (hunk 1 above)
Comment (plroebuck) [reply lib/growl.js]: I was referring to the JSDoc (L91-L94 in original) more so than the function arguments themselves.

**Pull request merged**